### PR TITLE
Remove null-checks for WindowManagerState

### PR DESCRIPTION
**Pull request by msisov**
Two fixups that remove added null-checks for WindowManagerState.

### CODE REVIEW

File: services/ui/ws/display.cc [hunk 1]
```diff
@@ -58,19 +58,17 @@ Display::~Display() {
...
-        ClientWindowId root_id;
-        if (tree->IsWindowKnown(root, &root_id))
-          tree->DeleteWindow(root_id);
-
-        // Destroy the tree once all the roots have been removed.
-        if (tree->roots().empty())
-          window_server_->DestroyTree(tree);
-      }
+    WindowTree* tree = display_root->window_manager_state()->window_tree();
+    ServerWindow* root = display_root->root();
```
Comment (fred-wang) [inline services/ui/ws/display.cc:62]: This is going to be a pain to merge. Maybe just remove the if() { ... } here and fix the indentation in a follow-up commit.

File: services/ui/ws/window_tree.cc [hunk 2]
```diff
@@ -302,7 +302,7 @@ void WindowTree::OnWmMoveDragImageAck() {
 bool WindowTree::SetCapture(const ClientWindowId& client_window_id) {
   ServerWindow* window = GetWindowByClientId(client_window_id);
   WindowManagerDisplayRoot* display_root = GetWindowManagerDisplayRoot(window);
-  if (display_root && !display_root->window_manager_state())
+  if (display_root)
```
Comment (fred-wang) [inline services/ui/ws/window_tree.cc]: For this and the one in WindowTree::ReleaseCapture there was no early return at all. I guess we can remove this display_root check unless to reduce the changeset. If that check turns out to be necessary, it should probably be upstream.

**Review by fred-wang — COMMENTED** [review]
Can you please name both commits "fixup! c++ / mojo changes for 'external window mode'" ?

File: services/ui/ws/display.cc (hunk 1 above)
Comment (msisov) [reply services/ui/ws/display.cc:62]: Why? There is no merge conflict

File: services/ui/ws/window_tree.cc [hunk 3]
```diff
@@ -302,8 +302,6 @@ void WindowTree::OnWmMoveDragImageAck() {
 bool WindowTree::SetCapture(const ClientWindowId& client_window_id) {
   ServerWindow* window = GetWindowByClientId(client_window_id);
   WindowManagerDisplayRoot* display_root = GetWindowManagerDisplayRoot(window);
-  if (display_root && !display_root->window_manager_state())
```
Comment (msisov) [inline services/ui/ws/window_tree.cc:305]: this is fixed!

File: services/ui/ws/display.cc (hunk 1 above)
Comment (msisov) [reply services/ui/ws/display.cc:62]: Divided to two different commits - remove if and fix indentation

**Review by fred-wang — APPROVED** [review]
lgtm. Please merge the PR without squashing.

**Review by tonikitoo — COMMENTED** [review]
lgtm

**Review by tonikitoo — COMMENTED** [review]
lgtm

**Review by tonikitoo — COMMENTED** [review]
lgtm

**Comment by tonikitoo** [comment]
feel free to merge them in the way you thing it works best for squashing.

**Pull request merged**